botocore/loaders.py

```diff
@@ -397,3 +439,23 @@ def _potential_locations(self, name=None, must_exist=False,
                         yield full_path
                     elif os.path.exists(full_path):
                         yield full_path
+
+
+class ExtrasProcessor(object):
+    """Processes data from extras files into service models."""
+    def process(self, service_data, extras_data):
+        """Processes data from a list of loaded extras files into a model
+
+        :type service_data: dict
+        :param service_data: The service model to load all the extras into.
+
+        :type extras_data: iterable of dict
```

Comment:
Based on the name of ``extras_data``, I would have expected this to be a dict because ``service_data`` is a dict. I wonder if it makes sense to rename this to something else? Something that makes it seem like a list or maybe making it ``*extra_data`` in the signature?